tools/autograd/templates/VariableType.cpp

```diff
@@ -65,15 +65,36 @@ Tensor & VariableType::checked_unpack(const Tensor & t, const char * name, int p
 {
  if(!t.defined()) {
    runtime_error("Expected a Tensor of type %s but found an undefined Tensor for argument #%d '%s'",
-     toString(),pos,name);
+     toString(), pos, name);
  }
  if (&t.type() == this) {
    return static_cast<VariableImpl*>(t.pImpl)->data;
  }
  runtime_error("Expected object of type %s but found type %s for argument #%d '%s'",
-   toString(),t.type().toString(),pos,name);
+   toString(),t.type().toString(), pos, name);
+}
+
+std::vector<at::Tensor> VariableType::checked_unpack(const at::TensorList &tl, const char *name, int pos) const {
+ std::vector<at::Tensor> ret(tl.size());
+ for (size_t i = 0; i < tl.size(); ++i) {
+   const auto &t = tl[i];
+   if(!t.defined()) {
+     runtime_error("Expected a Tensor of type %s but found an undefined Tensor at position #%d "
+                   "for TensorList argument #%d '%s'",
+                   toString(), i, pos, name);
+   }
+   if (&t.type() == this) {
+     ret[i] = static_cast<VariableImpl*>(t.pImpl)->data;
+   } else {
+   runtime_error("Expected object of type %s but found type %s at position #%d "
+                 "for TensorList argument #%d '%s'",
```

Comment:
We shouldn't mention TensorList in error messages. Think about the user who has no idea about the C code. It's not helpful

Reply:
The user doesn't know about undefined tensors either, but those are already mentioned in error messages :).

Maybe I'll just say "for iterable argument" so the position number still makes sense.

Reply:
True, however I have no idea if it's possible to even get an undefined tensor in Python (but let's keep the check to be sure). Passing incorrect arguments in sequences is common